### PR TITLE
feat!(deps): bump driver to 6.7.0 MONGOSH-1790

### PR DESCRIPTION
**Pull request by mabaasit**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -74,7 +74,7 @@
     "gen-esm-wrapper": "^1.1.0",
     "husky": "^8.0.3",
     "mocha": "^10.2.0",
-    "mongodb": "^5.4.0",
+    "mongodb": "^6.7.0",
```
Comment (addaleax) [inline package.json:77]: This is already close to the maximal breaking change we could do for this work, we might as well remove `abort-controller` too.

File: src/types.ts [hunk 2]
```diff
@@ -94,32 +94,23 @@ export interface IdPServerResponse {
...
+ * A copy of the Node.js driver's `OIDCCallbackParams` using `OIDCAbortSignal` instead of `AbortSignal`
  * @public
  */
-export interface OIDCCallbackContext {
+export interface OIDCCallbackParams {
   refreshToken?: string;
-  timeoutSeconds?: number;
-  timeoutContext?: OIDCAbortSignal;
-  version: number;
+  timeoutContext: OIDCAbortSignal;
```
Comment (addaleax) [inline src/types.ts:102]:
```suggestion
  timeoutContext?: OIDCAbortSignal;
```

Or is there a specific reason why this needs to be required?

File: src/plugin.spec.ts [hunk 3]
```diff
@@ -57,15 +56,13 @@ async function fetchBrowser({ url }: OpenBrowserOptions): Promise<void> {
...
-  if (typeof abortSignal === 'number') clientInfo.timeoutSeconds = abortSignal;
-  else if (abortSignal) clientInfo.timeoutContext = abortSignal;
-  return plugin.mongoClientOptions.authMechanismProperties.REQUEST_TOKEN_CALLBACK(
-    oidcParams,
-    clientInfo
-  );
+  abortSignal?: OIDCAbortSignal
+): ReturnType<OIDCCallbackFunction> {
+  return plugin.mongoClientOptions.authMechanismProperties.OIDC_HUMAN_CALLBACK({
+    timeoutContext: abortSignal ?? new AbortController().signal,
```
Comment (addaleax) [inline src/plugin.spec.ts:62]:
```suggestion
    timeoutContext: abortSignal,
```

**Review by addaleax — APPROVED** [review]

File: src/types.ts (hunk 2 above)
Comment (mabaasit) [reply src/types.ts:102]: driver has marked it as required prop. if we don't have to align this bit, i'll leave it as optional

File: package.json (hunk 1 above)
Comment (mabaasit) [reply package.json:77]: you mean our `OIDCAbortSignal`, right?

Comment (addaleax) [reply package.json:77]: No – I mean `abort-controller` 🙂

File: src/types.ts (hunk 2 above)
Comment (addaleax) [reply src/types.ts:102]: We can always be more accepting than the driver

File: package.json (hunk 1 above)
Comment (mabaasit) [reply package.json:77]: ah, i noticed it in deps! will remove it

Comment (mabaasit) [reply package.json:77]: removed in [67ede2b](https://github.com/mongodb-js/oidc-plugin/pull/191/commits/67ede2b35a9def9bcfb528ec10e4333ffa30004f)

**Review by addaleax — APPROVED** [review]

**Pull request merged**